remove_debugging_statements.yaml

```diff
@@ -0,0 +1,47 @@
+# These optional rules flag or remove various debugging statements.
+#
+# All rules in this file have the tag `no-debug`.
+#
+
+rules:
+- id: remove-breakpoint
+  description: Remove breakpoints from production code
+  explanation: Breakpoints shouldn't be used in production code.
+  pattern: breakpoint()
+  replacement: ''
+  tests:
+  - match: breakpoint()
+  - no-match: breakpoint("sth")
+  tags:
+  - no-debug
+
+- id: remove-pdb-set_trace
+  description: Remove `pdb.set_trace` calls from production code
+  explanation: Breakpoints shouldn't be used in production code.
+  pattern: pdb.set_trace()
+  replacement: ''
+  tests:
+  - match: |
+      import pdb
+      pdb.set_trace()
+  - match: |
+      import pdb as pdb_debug
+      pdb_debug.set_trace()
+  tags:
+  - no-debug
+
+- id: flag-print
+  description: Remove `print` statements from production code
+  explanation: |
+    `print` statements shouldn't be used in production code.
+    Use:
+
+    * a CLI framework for output
+    * `logger.debug()` for debugging information.
+  pattern: print(...)
```

Comment:
Interesting... why not provide a `replacement: ''` here as well?

Reply:
I'm not sure what the replacement should be:

* a `logger.debug(...)` with the same arguments?
* nothing?

In the [docs](https://docs.sourcery.ai/Guides/Custom-Rules/Recipes/remove-debugging-statements/), we mention multiple possibilities. Here, I tried to keep it general. :slightly_smiling_face:

Reply:
Ah, alright, that makes sense!